ebml/EbmlCrc32.h

```diff
@@ -94,7 +95,7 @@ DECLARE_EBML_BINARY(EbmlCrc32)
     void ResetCRC();
     void UpdateByte(binary b);
 
-    static const uint32 m_tab[256];
+    static const std::array<uint32, 256> m_tab;
```

Comment:
What is the performance impact of this change ? Direct access to a buffer in memory seems a lot faster than using class methods to do the same. Or when compiled using `.at()` and `.data()` end up with similar generated code ?

Reply:
There is none. std::array is guarenteed to be >= fast as C arrays.

.at() in release mode is the same as []. In debug mode, it does some checks AFAIK.

Reply:
This is true with gcc and clang, this does not appear to be the case with MSVC. Even with `-Os` it does a call for `.at()` and `[]`.
https://stackoverflow.com/a/52805924

https://godbolt.org/z/PrMnWP vs https://godbolt.org/z/zMMEKc

Reply:
Unfortunate. I have no experience with MSVC.

Reply:
The `-Os` flag [may not be the best](https://docs.microsoft.com/en-us/cpp/build/reference/compiler-options-listed-by-category) one. With `-O2 -GS-` the difference is even more dramatic:
https://godbolt.org/z/vEx5Ys vs https://godbolt.org/z/nKrxPT

Reply:
Just looked at godbolt. clang generates the same assembly with .at and [], even with libc++.

There's something wrong with MSVC.

Reply:
The StackOverflow post also mentioned ICC which was broken but I tried the latest and it seems to generate the same code as C-style arrays. I tried looking at the [MS STL](https://github.com/microsoft/STL/blob/main/stl/inc/array) and playing with some compilation parameter https://godbolt.org/z/aceTa9. It doesn't do any STL `call` so it's better but it's not matching the C-style generated code, which might be even more optimized.

I'm OK with your change. As long as there is no `call` there I suppose they already do some optimizations. I haven't tested the actual speed.